default/python/AI/MilitaryAI.py

```diff
@@ -421,6 +424,22 @@ def _calculate_threat(self):
     def _take_any(self):
         return self.assigned_rating > 0
 
+    def _planet_threat_multiplier(self):
+        # to the extent that enemy planetary defenses are bolstered by fleet defenses, a smaller portion of our
+        # attacks will land on the planet and hence we need a greater portion of planet-effective attacks.  One
+        # desired characteristic of the following planet_threat_multiplier is that if the entire local threat is
+        # due to planetary threat then we want the multiplier to be unity.  Furthermore, the more enemy ships are
+        # present, the smaller proportion of our attacks would be directed against the enemy planet.  The following is
+        # just one of many forms of calculation that might work reasonably.
+        # TODO: assess and revamp the planet_threat_multiplier calculation
+        return  ((self._enemy_ship_count() + self._local_threat()/self._planet_threat())**0.5
+                 if self._planet_threat() > 0 else 1.0)
+
+    def _allocation_vs_planets(self):
+        return CombatRatingsAI.rating_needed(
+            self.safety_factor*self._planet_threat_multiplier()*self._planet_threat(),
+            self.assigned_rating_vs_planets)
```

Comment:
It was intentional to only do it for the TargetAllocator (and its further subclasses), yes.  

For the two defense type allocators, the most important timing for planning is before the enemy forces actually move into the system, and the current form of calculation I have here just won't help then.  Also the AI will already also have at least one planet of their own there, generally contributing at least something to the battle to counterbalance the other side having a planet present.  It's just a more complicated scenario I wasn't anxious to jump into tinkering with. 

However, planning some more flexibility for the future, I suppose this could nearly all go into the base Allocator class except with _planet_threat_multiplier() being a flat unity there, and then this more complicated calc for _planet_threat_multiplier() would override that for the TargetAllocators.  Does that sound better to you?